model_analyzer/result/result_manager.py

```diff
@@ -98,46 +105,31 @@ def add_server_data(self, data):
         self._state_manager.set_state_variable('ResultManager.server_only_data',
                                                data)
 
-    def add_run_config_measurement(self, run_config, run_config_measurement):
+    def add_run_config_measurement(
+            self, run_config, run_config_measurement: RunConfigMeasurement):
         """
-        This function adds model inference
-        measurements to the required result
-
-        Parameters
-        ----------
-        run_config : RunConfig
-            Contains the parameters used to generate the measurment
-        run_config_measurement: RunConfigMeasurement
-            the measurement to be added
+        Add measurement to individual result heap,
+        global result heap and results class
         """
+        model_name = run_config.models_name()
 
-        # Get reference to results state and modify it
-        results = self._state_manager.get_state_variable(
-            'ResultManager.results')
+        run_config_result = RunConfigResult(
+            model_name=model_name,
+            run_config=run_config,
+            comparator=self._run_comparators[model_name],
+            constraints=self._run_constraints[model_name])
 
-        results.add_run_config_measurement(run_config, run_config_measurement)
+        run_config_measurement.set_metric_weightings(
```

Comment:
I don't love that we have to reach into private variables here.

also, is there ever a time we would want to set metric weights and not model weights, or vice-versa?

Reply:
Yeah, I agree, but this is an issue with the RunConfigResultComparator class. I'll add a public get function to the class.

Reply:
Added public class.